*** CPUs/cv32e40p/cxxrtl_cv32e40p.cpp.orig	2026-10-08 01:57:34.481836403 +0200
--- CPUs/cv32e40p/cxxrtl_cv32e40p.cpp	2026-10-08 01:58:51.418780925 +0200
***************
*** 1943,1946 ****
--- 1943,1955 ----
  	// cell \_3576_
  	p_u__top_2e_core__i_2e_load__store__unit__i_2e_data__addr__int = symb_mux(p_u__top_2e_core__i_2e_id__stage__i_2e_prepost__useincr__ex__o.curr, p___624__, p_u__top_2e_core__i_2e_alu__operand__a__ex.curr);
+     if (p_u__top_2e_core__i_2e_data__req__ex.curr.data[0] and Extract(31, 20, *p_u__top_2e_core__i_2e_load__store__unit__i_2e_data__addr__int.node).nature != CONST) {
+         std::cout << "Concretizing lsu MSBs, because there is a read this cycle" << std::endl;
+         p_u__top_2e_core__i_2e_load__store__unit__i_2e_data__addr__int.setNode(&Concat(
+             constant(p_u__top_2e_core__i_2e_load__store__unit__i_2e_data__addr__int.get<uint32_t>() >> 20, 12),
+             Extract(19, 2, *p_u__top_2e_core__i_2e_load__store__unit__i_2e_data__addr__int.node),
+             constant(p_u__top_2e_core__i_2e_load__store__unit__i_2e_data__addr__int.get<uint32_t>() & 0b11, 2)
+         ));
+         p_u__top_2e_core__i_2e_load__store__unit__i_2e_data__addr__int.debug_assert();
+     }
  	// \parallel_case: 1
  	// \full_case: 1
